src/config.rs

```diff
@@ -158,6 +163,19 @@ impl Config {
 
         false
     }
+
+    pub(crate) fn rule_config<'a, T>(&'a self, ident: &str) -> Option<T>
```

Comment:
I _think_ this should be `Result<T>`, with the following states:

* If `policy` is `Value::Null`, we should return `Ok(T::default())`
* If `policy` loads correctly, we should return `Ok(policy)`
* If `policy` fails to deserialize, we should return `Err(...)`

That way we present an appropriate hard error to the user when their configuration is invalid, rather than using the default configuration (which is probably not that they intended) 🙂

Reply:
(This also requires a `T: Default` constraint, but IMO that's fine!)